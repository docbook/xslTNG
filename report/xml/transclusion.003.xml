<article xmlns="http://docbook.org/ns/docbook"
         xmlns:trans="http://docbook.org/ns/transclusion"
         xmlns:xi='http://www.w3.org/2001/XInclude'
	 xmlns:xlink="http://www.w3.org/1999/xlink">
<title>Transclusion 003</title>

<para>In this document, all transcluded IDs are resolved with the ‘suffix’ fixup
and the default linkscope which is ‘global’.</para>

<para>In a document where the same fragment is included (for example,
with XInclude) more than once, any ID values in that fragment will
be duplicated in the resulting document. Duplicate IDs should be avoided.
</para>

<para>The DocBook <citetitle
xlink:href="https://docbook.org/docs/transclusion/">Transclusion</citetitle>
specification is intended to address this problem by providing “fixup”
mechanisms.</para>

<para>In this document, we use the following <tag linkend="caution">caution</tag>
element as an example:</para>

<xi:include href="tran-caution.xml" trans:idfixup="suffix" trans:suffix="one"/>

<para>Here it is <link linkend="caution">repeated</link>:</para>

<xi:include href="tran-caution.xml" trans:idfixup="suffix" trans:suffix="two"/>

<section>
<title>Subsection One</title>

<para>Here it is <link linkend="caution">repeated</link> in a subsection:</para>

<xi:include href="tran-caution.xml" trans:idfixup="suffix" trans:suffix="three"/>

</section>

<section>
<title>Subsection Two</title>

<para>Here it is <link xlink:href="#caution">repeated</link> in another subsection:</para>

<xi:include href="tran-caution.xml" trans:idfixup="suffix" trans:suffix="four"/>

</section>
</article>
