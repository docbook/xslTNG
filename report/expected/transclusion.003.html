<!DOCTYPE html><html xmlns="http://www.w3.org/1999/xhtml" class="no-js"><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"/><script>(function(H){H.className=H.className.replace(/\bno-js\b/,'js')})(document.documentElement)</script><title>Transclusion 003</title><meta name="viewport" content="width=device-width, initial-scale=1.0"/><link href="https://purl.org/dc/elements/1.1/" rel="schema.dc"/><meta content="2011-04-22T17:02:00-06:00" name="dc.modified"/><meta content="DocBook xslTNG" name="generator"/><link href="./css/docbook.css" rel="stylesheet" media="screen"/></head><body class="home"><nav class="top"></nav><main><article class="article component"><header><h1>Transclusion 003</h1></header><div class="list-of-titles"><div class="lot toc"><div class="title">Table of Contents</div><ul class="toc"><li><a href="#R_s1"><span class="label">1</span><span class="sep">. </span>Subsection One</a></li><li><a href="#R_s2"><span class="label">2</span><span class="sep">. </span>Subsection Two</a></li></ul></div></div><p>In this document, all transcluded IDs are resolved with the ‘suffix’ fixup
and the default linkscope which is ‘global’.</p><p>In a document where the same fragment is included (for example,
with XInclude) more than once, any ID values in that fragment will
be duplicated in the resulting document. Duplicate IDs should be avoided.
</p><p>The DocBook <i class="citetitle"><a href="https://docbook.org/docs/transclusion/">Transclusion</a></i>
specification is intended to address this problem by providing “fixup”
mechanisms.</p><p>In this document, we use the following <code class="tag tag-element"><a href="#cautionone">caution</a></code>
element as an example:</p><div id="cautionone" class="admonition caution"><div><div class="icon">⚠</div><div class="body"><header><div class="title">Transclusion Example Caution</div></header><div>

<p>This caution has the <code>xml:id</code> value ‘<code>caution</code>’.
If it is simple transcluded more than once into a document, it will
introduce duplicate IDs in the document.
</p>
</div></div></div></div><p>Here it is <a href="#cautionone" class="link">repeated</a>:</p><div id="cautiontwo" class="admonition caution"><div><div class="icon">⚠</div><div class="body"><header><div class="title">Transclusion Example Caution</div></header><div>

<p>This caution has the <code>xml:id</code> value ‘<code>caution</code>’.
If it is simple transcluded more than once into a document, it will
introduce duplicate IDs in the document.
</p>
</div></div></div></div><section id="R_s1" class="section"><header><h2><span class="label">1</span><span class="sep">. </span>Subsection One</h2></header><p>Here it is <a href="#cautionone" class="link">repeated</a> in a subsection:</p><div id="cautionthree" class="admonition caution"><div><div class="icon">⚠</div><div class="body"><header><div class="title">Transclusion Example Caution</div></header><div>

<p>This caution has the <code>xml:id</code> value ‘<code>caution</code>’.
If it is simple transcluded more than once into a document, it will
introduce duplicate IDs in the document.
</p>
</div></div></div></div></section><section id="R_s2" class="section"><header><h2><span class="label">2</span><span class="sep">. </span>Subsection Two</h2></header><p>Here it is <a href="#cautionone" class="link">repeated</a> in another subsection:</p><div id="cautionfour" class="admonition caution"><div><div class="icon">⚠</div><div class="body"><header><div class="title">Transclusion Example Caution</div></header><div>

<p>This caution has the <code>xml:id</code> value ‘<code>caution</code>’.
If it is simple transcluded more than once into a document, it will
introduce duplicate IDs in the document.
</p>
</div></div></div></div></section></article></main><nav class="bottom"></nav></body></html>